### ISSUE TITLE
`object_name_linter()` complains about ns-hooks (e.g. `.onLoad`)

### ISSUE THREAD

**Issue by salim-b** [jimhester/lintr#500]
Reprex:

```r
example_file <- tempfile()

readr::write_file(".onLoad <- function(libname, pkgname) {\n  # some action\n}\n",
                  path = example_file)

lintr::lint(filename = example_file, linters = lintr::object_name_linter())
```

R's [hooks for namespace events](https://rdrr.io/r/base/ns-hooks.html) are CamelCase and you can't change that AFAIK, so lintr complaining about

```
Variable and function name style should be snake_case.
```

is pointless.

**Comment by AshesITR** [comment]
Thanks for reporting this.

The following function names are special in this regard:

```r
.onLoad(libname, pkgname)
.onAttach(libname, pkgname)
.onUnload(libpath)
.onDetach(libpath)
.Last.lib(libpath)
```

Additionally, `.Rprofile` files may also define `.First` and `.Last`.

I think a good place to add that would be in a new `is_special_function()` for use at https://github.com/jimhester/lintr/blob/83a73fc8698bcd2dbc82c9a12d2b4e929a811a73/R/object_name_linters.R#L129-L133

@jimhester What do you think about this?

**Comment by jimhester** [comment]
Yeah, that sounds like a good plan to special case these names.
